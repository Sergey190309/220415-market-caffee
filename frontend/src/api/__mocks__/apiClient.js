import {
  // resolveDataTechInPost,
  resolveDataTechInGet,
  // resolveDataPutTextContent
} from '../../testAxiosConstants'

// export const refreshTokenAxiosClient = {
//   put: ()=> Promise.resolve()
// }

export const authTextAxiosClient = {
  put: jest.fn(() => Promise.resolve({ data: {} }))
}

export const techTextAxiosClient = {
  // post: () => Promise.resolve(resolveDataTechInPost),
  get: () => Promise.resolve(resolveDataTechInGet),
  defaults: {
    headers: {
      common: ['Authorization']
    }
  }
}

export const setAxiosCommonLng = jest.fn()
export const setAxiosTechToken = jest.fn()
export const setAxiosAuthAccessToken = jest.fn()
export const setAxiosAuthRefreshToken = jest.fn()
