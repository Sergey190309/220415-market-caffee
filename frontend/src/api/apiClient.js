import axios from 'axios'
// import i18next from 'i18next';

const baseURL = 'http://127.0.0.1:5000'
const headers = {
  // Accept: 'application/json',
  'Content-Type': 'application/json',
  common: {
    // 'Accept-Language': 'ru'
    'Accept-Language': 'en'
  }
}

export const techAxiosClient = axios.create({
  /**
   * used for tech initial update - tech token, sessions, etc.
   */
  baseURL: baseURL,
  headers: headers
})

export const pixAxiosClient = axios.create({
  /**
   * user for operations with pictures
   */
  baseURL: baseURL,
  headers: {
    ...headers,
    'Content-Type': 'multipart/form-data'

    // 'Access-Control-Allow-Origin': '*',
    // 'Access-Control-Allow-Methods': 'POST, GET, OPTIONS, PUT, DELETE',
    // 'Access-Control-Allow-Headers': 'Origin, Content-Type, Accept, Authorization, X- Request-With'
  },
  responseType: 'arraybuffer',
  responseEncoding: 'binary'
})

export const authAxiosClient = axios.create({
  /**
   * used for authorisation
   */
  baseURL: baseURL,
  headers: headers
})

export const axiosCommonLng = lng => {
  // console.log('apiClient, axiosCommonLng, lng ->', lng)
  techAxiosClient.defaults.headers.common['Accept-Language'] = lng
  pixAxiosClient.defaults.headers.common['Accept-Language'] = lng
  authAxiosClient.defaults.headers.common['Accept-Language'] = lng
}

export const axiosCommonToken = (token, axiosInstance = techAxiosClient) => {
  // console.log('axiosCommonToken, headers ->', axiosInstance.defaults.headers.common['Authorization'])
  axiosInstance.defaults.headers.common.Authorization = `Bearer ${token}`
  pixAxiosClient.defaults.headers.common.Authorization = `Bearer ${token}`
}

// export default axiosClient;
