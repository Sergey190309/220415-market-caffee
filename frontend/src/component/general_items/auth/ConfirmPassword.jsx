import React from 'react'
import PropTypes from 'prop-types'

const ConfirmPassword = props => {
  return (
    <div>ConfirmPassword</div>
  )
}

ConfirmPassword.propTypes = {}

export default ConfirmPassword