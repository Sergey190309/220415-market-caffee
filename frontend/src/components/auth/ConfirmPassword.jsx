import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import PropTypes from 'prop-types'
import { Container, Segment, Icon, Header, Grid, Button } from 'semantic-ui-react'
import { Formik } from 'formik'
import { Form, Input, SubmitButton } from 'formik-semantic-ui-react'
import * as Yup from 'yup'
import { useTranslation } from 'react-i18next'

import {
  closeModal,
  confirmPasswordStart,
  confirmPasswordModalClosed,
  setMessage,
  deviceSelector,
  authSelector
} from '../../redux/slices'
import Alert from '../layout/Alert'

import { positiveColor, warningColor } from '../../utils/colors'

export const formStructure = {
  password: 'qwerty'
}

export const confirmPasswordSchema = t =>
  Yup.object().shape({
    [Object.keys(formStructure)[0]]: Yup.string()
      .required(t('confirmpassword.errors.required'))
  })

const ConfirmPassword = ({
  initValues,
  confirmPasswordSchema,
  closeModal,
  confirmPasswordStart,
  confirmPasswordModalClosed,
  setMessage
}) => {
  const dispatch = useDispatch()
  const { isConfirmedPassword } = useSelector(authSelector)
  const { message } = useSelector(deviceSelector)
  const { t } = useTranslation('auth')

  useEffect(() => {
    if (isConfirmedPassword) {
      dispatch(closeModal())
      dispatch(confirmPasswordModalClosed())
    }
  }, [isConfirmedPassword])

  const onSubmit = (formData, { setSubmitting }) => {
    // console.log('component, auth, ConfirmPassword, formData ->', formData)
    dispatch(confirmPasswordStart(formData))
    setSubmitting(false)
  }

  return (
    <Container
      data-testid='whole-component'
      fluid textAlign='center'
    >
      <Alert />
      <Grid
        textAlign='center'
        style={{ height: '50vh' }}
        verticalAlign='middle'
      >
        <Grid.Column style={{ maxWidth: 500 }}>
          <Header as='h2' textAlign='center' color={positiveColor}>
            <Segment.Inline>
              <Icon name='utensils' size='large' />
              {t('confirmpassword.header')}
            </Segment.Inline>
          </Header>

          <Formik
            initialValues={initValues}
            validationSchema={confirmPasswordSchema(t)}
            onSubmit={onSubmit}
          >
            {({ isSubmitting }) => (
              <Form size='large'>
                <Segment color={positiveColor} stacked>
                  <Header content={message} />
                  <Input
                    id='input-password'
                    data-testid='input-password'
                    name='password'
                    type='password'
                    inputLabel={t('confirmpassword.labels.password')}
                    // inputLabel={{ color: color, content: 'Password' }}
                    icon='key'
                    placeholder={t('confirmpassword.placeHolders.password')}
                    autoComplete='on'
                    errorPrompt
                  />
                  <Button.Group widths='1'>
                    <SubmitButton
                      basic
                      color={positiveColor}
                      size='large'
                      content={t('confirmpassword.buttons.confirm')}
                      // disabled={isSubmitting}
                    />
                    <Button.Or text={t('confirmpassword.buttons.or')} />
                    <Button
                      basic
                      color={warningColor}
                      size='large'
                      content={t('confirmpassword.buttons.cancel')}
                      type='button'
                      onClick={() => {
                        // console.log('onClick, closeMocal')
                        dispatch(setMessage(''))
                        dispatch(closeModal())
                      }}
                    />
                  </Button.Group>
                </Segment>
              </Form>
            )}
          </Formik>
        </Grid.Column>
      </Grid>
    </Container>
  )
}

ConfirmPassword.defaultProps = {
  initValues: formStructure,
  confirmPasswordSchema,
  closeModal,
  confirmPasswordStart,
  confirmPasswordModalClosed,
  setMessage
}

ConfirmPassword.propTypes = {
  initValues: PropTypes.object.isRequired,
  confirmPasswordSchema: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  confirmPasswordStart: PropTypes.func.isRequired,
  confirmPasswordModalClosed: PropTypes.func.isRequired,
  setMessage: PropTypes.func.isRequired
}

export default ConfirmPassword
