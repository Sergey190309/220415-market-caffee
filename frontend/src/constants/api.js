// export const baseURL = 'http://localhost:5000/'
export const baseURL = 'http://localhost/'
// Below shoud be set while running in docker container.
// export const baseURL = '/'