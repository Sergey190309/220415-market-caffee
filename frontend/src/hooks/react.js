import {useState, useEffect, useRef, useCallback } from 'react'

export const useAppState = useState
export const useAppEffect = useEffect
export const useAppRef = useRef
export const useAppCallback = useCallback
