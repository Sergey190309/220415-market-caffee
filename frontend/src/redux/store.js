import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import createSagaMiddleware from 'redux-saga'
// import reducer from './reducers';
import reducer from './slices'
import rootSaga from './saga'
// import { rootReducer } from './reducers';

const initialState = {}
const sagaMiddleware = createSagaMiddleware()

const middleware = [...getDefaultMiddleware({ thunk: false }), sagaMiddleware]
// const middleware = [sagaMiddleware, thunk];

const store = configureStore({
  reducer,
  middleware,
  initialState
})

sagaMiddleware.run(rootSaga)

export default store
