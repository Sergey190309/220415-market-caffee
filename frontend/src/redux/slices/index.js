import { combineReducers } from '@reduxjs/toolkit'
// import { combineReducers } from 'redux';
// import alerts from '../reducers/alert';
import alerts from './alerts'
// import logIn from '../reducers/auth';
import auth from './auth'
import device from './device'
// import device from '../reducers/device';
import lng from './lng'
// import lng from '../reducers/lng';
import structure from './structure'
import tech from './tech'

export const rootReducer = combineReducers({
  alerts,
  auth,
  // logIn,
  device,
  lng,
  structure,
  tech
})

export { startAlert, removeAlert } from './alerts'
export {
  setState as authSetState,
  signUpStart,
  signUpSuccess,
  signUpFail,
  logInStart,
  logInSuccess,
  logInFail,
  logOut,
  signUpModalClosed,
  logInModalClosed,
  authSelector
} from './auth'
export { deviceSelector, setDeviceSize, openModal, closeModal, setEditable } from './device'
export { lngSwitch, lngSelector } from './lng'
export {
  setState as structureSetState,
  structureSelector,
  structureStart,
  structureSuccess,
  structureFail
} from './structure'
export {
  techSelector,
  startInitLoading,
  initLoadingSuccess,
  startTechIn,
  techInSuccess,
  techInFail,
  startLngs,
  lngsSuccess,
  lngsFail,
  startI18n,
  i18nInitiated,
  i18nSuccess,
  i18nFail
} from './tech'
export default rootReducer
