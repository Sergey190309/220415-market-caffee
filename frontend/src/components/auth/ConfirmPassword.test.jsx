<<<<<<< HEAD
describe('ConfirmPassword testing', () => {
  test('dummy', () => {
=======
import React from 'react'
import { Provider } from 'react-redux'
import { useTranslation } from 'react-i18next'
>>>>>>> dev210823

import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import store from '../../redux/store'
import ConfirmPassword, { confirmPasswordSchema, formStructure } from './ConfirmPassword'
import { clearAlerts, closeModal, confirmPasswordModalClosed, confirmPasswordStart, setMessage, startAlert } from '../../redux/slices'
import { refreshTokenAxiosClient as mockRefreshTokenClient } from '../../api/apiClient'
import { resolveDataConfirmPassword } from '../../testAxiosConstants'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: key => key
    // i18n: { changeLanguage: jest.fn() },
  })
}))

describe('ConfirmPassword component testing', () => {
  const initValues = {
    password: 'qwerty'
  }
  describe('Non react part', () => {
    test('form structure', () => {
      expect(formStructure).toEqual(initValues)
    })
    test('Yup schema', () => {
      const { t } = useTranslation()
      const result = confirmPasswordSchema(t)
      expect(result.fields.password.type).toBe('string')
      // console.log('auth, confirmPassword.test, result ->', result)
    })
  })
  describe('Component testing', () => {
    const alertPayload = {
      message: 'mockMessage',
      alertType: 'info',
      timeout: 3000,
      id: 'mockId'
    }
    const testProps = {
      initValues,
      confirmPasswordSchema: jest.fn(),
      closeModal: jest.fn(),
      confirmPasswordStart: jest.fn(),
      confirmPasswordModalClosed: jest.fn(),
      setMessage: jest.fn()
    }
    beforeEach(() => {
      jest.resetAllMocks()
    })
    describe('appearance', () => {
      test('it exists and can be rendered (snapshot)', () => {
        render(
          // const { container } = render(
          <Provider store={store}>
            <ConfirmPassword />
          </Provider>
        )
        const wholeComponent = screen.getByTestId('whole-component')
        expect(wholeComponent).toMatchSnapshot()
      })
      test('it has all appropriate elements with alert', () => {
        store.dispatch(startAlert(alertPayload))
        render(
          // const { container } = render(
          <Provider store={store}>
            <ConfirmPassword />
          </Provider>
        )
        const alertHeader = screen.getByRole('heading', { name: alertPayload.message })
        expect(alertHeader).toHaveClass('ui teal header', { exact: true })
        expect(alertHeader).toHaveTextContent(alertPayload.message)
        expect(alertHeader.firstChild).toHaveClass('hand point right outline icon', { exact: true })
        const header = screen.getByRole('heading', { name: 'confirmpassword.header' })
        expect(header).toHaveClass('ui teal center aligned header', { exact: true })
        expect(header.firstChild.firstChild).toHaveClass('utensils large icon', { exact: true })
        const input = screen.getByTestId('input-password')
        expect(input.children).toHaveLength(3)
        expect(input.children[0]).toHaveClass('ui label label', { exact: true })
        expect(input.children[1]).toHaveAttribute('id', 'input-password')
        expect(input.children[2]).toHaveClass('key icon', { exact: true })
        const confirmButton = screen.getByRole('button', { name: 'confirmpassword.buttons.confirm' })
        expect(confirmButton).toHaveClass('ui teal large basic button', { exact: true })
        expect(confirmButton).toHaveAttribute('id', 'formik-semantic-ui-react-submit-button')
        const cancelButton = screen.getByRole('button', { name: 'confirmpassword.buttons.cancel' })
        expect(cancelButton).toHaveClass('ui orange large basic button', { exact: true })
        // console.log('auth, confirmPassword.test, element ->', input.children.length)
        // screen.debug(alertHeader)
      })

      test('input field contains value accrding props', () => {
        const actualProps = {
          ...testProps,
          initValues: {
            password: 'mock password'
          }
        }
        store.dispatch(clearAlerts())
        render(
          <Provider store={store}>
            <ConfirmPassword {...actualProps} />
          </Provider>
        )
        const input = screen.getByTestId('input-password')
        expect(input.childElementCount).toBe(3)
        expect(input.children[1]).toHaveValue(actualProps.initValues.password)
        // console.log('auth, confirmPassword.test, element ->', input.children[1].value)
        // screen.debug(input)
      })
    })
    describe('controls behavior ', () => {
      let actualProps

      beforeEach(() => {
        jest.resetAllMocks()
        actualProps = {
          ...testProps,
          confirmPasswordStart: jest.fn(() => ({ type: confirmPasswordStart.type })),
          // confirmPasswordStart: jest.fn().mockReturnValue({ type: confirmPasswordStart.type }),
          closeModal: jest.fn(() => ({ type: closeModal.type })),
          confirmPasswordModalClosed: jest.fn(() => ({ type: confirmPasswordModalClosed.type })),
          setMessage: jest.fn(() => ({ type: setMessage.type, payload: 'mockMessage' }))
        }
      })
      test('input', async () => {
        const inputText = 'input text'
        render(
          <Provider store={store}>
            <ConfirmPassword {...testProps} />
          </Provider>
        )
        const input = screen.getByTestId('input-password').children[1]
        userEvent.clear(input)
        userEvent.type(input, inputText)
        await waitFor(() => {
          expect(input).toHaveValue(inputText)
        })
        // console.log('auth, confirmPassword.test, element ->', input.value)
        // screen.debug(input)
      })
      test('confirm button', async () => {
        mockRefreshTokenClient.put.mockImplementation(() => Promise.resolve({ data: resolveDataConfirmPassword }))

        render(
          <Provider store={store}>
            <ConfirmPassword {...actualProps} />
          </Provider>
        )
        const confirmButton = screen.getByRole('button', { name: 'confirmpassword.buttons.confirm' })
        userEvent.click(confirmButton)
        await waitFor(() => {
          expect(actualProps.confirmPasswordStart).toHaveBeenCalledTimes(1)
          expect(actualProps.confirmPasswordStart).toHaveBeenCalledWith(initValues)
          expect(actualProps.closeModal).toHaveBeenCalledTimes(1)
          expect(actualProps.confirmPasswordModalClosed).toHaveBeenCalledTimes(1)
        })
        // console.log('auth, confirmPassword.test, element ->', confirmButton.value)
        // screen.debug(confirmButton)
      })
      test('cancel button', async () => {
        mockRefreshTokenClient.put.mockImplementation(() => Promise.resolve({ data: resolveDataConfirmPassword }))

        render(
          <Provider store={store}>
            <ConfirmPassword {...actualProps} />
          </Provider>
        )
        const cancelButton = screen.getByRole('button', { name: 'confirmpassword.buttons.cancel' })
        userEvent.click(cancelButton)
        await waitFor(() => {
          // expect(actualProps.confirmPasswordStart).toHaveBeenCalledTimes(1)
          expect(actualProps.setMessage).toHaveBeenCalledTimes(1)
          expect(actualProps.setMessage).toHaveBeenCalledWith('')
          expect(actualProps.closeModal).toHaveBeenCalledTimes(1)
          // expect(actualProps.confirmPasswordModalClosed).toHaveBeenCalledTimes(1)
        })
        // console.log('auth, confirmPassword.test, element ->', cancelButton.value)
        // screen.debug(cancelButton)
      })
    })
  })
})
