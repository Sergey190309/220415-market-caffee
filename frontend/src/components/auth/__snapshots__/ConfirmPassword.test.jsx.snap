// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmPassword component testing Component testing appearance it exists and can be rendered (snapshot) 1`] = `
<div
  class="ui fluid center aligned container"
  data-testid="whole-component"
>
  <div
    class="ui center aligned middle aligned grid"
    style="height: 50vh;"
  >
    <div
      class="column"
      style="max-width: 500px;"
    >
      <h2
        class="ui teal center aligned header"
      >
        <div
          class="inline"
        >
          <i
            aria-hidden="true"
            class="utensils large icon"
          />
          confirmpassword.header
        </div>
      </h2>
      <form
        class="ui large form"
      >
        <div
          class="ui teal stacked segment"
        >
          <div
            class="ui header"
          >
            
          </div>
          <div
            class="field"
          >
            <div
              class="ui icon labeled input"
              data-testid="input-password"
            >
              <div
                class="ui label label"
                for="input-password"
                id="input-password-input-label"
              >
                confirmpassword.labels.password
              </div>
              <input
                aria-labelledby=" input-password-input-label"
                autocomplete="on"
                id="input-password"
                name="password"
                placeholder="confirmpassword.placeHolders.password"
                type="password"
                value="qwerty"
              />
              <i
                aria-hidden="true"
                class="key icon"
              />
            </div>
          </div>
          <div
            class="ui one buttons"
          >
            <div
              class="field"
            >
              <button
                class="ui teal large basic button"
                id="formik-semantic-ui-react-submit-button"
                type="submit"
              >
                confirmpassword.buttons.confirm
              </button>
            </div>
            <div
              class="or"
              data-text="confirmpassword.buttons.or"
            />
            <button
              class="ui orange large basic button"
              type="button"
            >
              confirmpassword.buttons.cancel
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
